networks/ethereum/wallet.csv

```diff
@@ -1,14 +1,28 @@
+safepal,!provider:safepal,FALSE,"[Browser Extension"", ""Hardware Wallet"",""iOS"",""Android""]",FALSE,TRUE,FALSE,,,,,,,,,,,,,"https://www.safepal.com/, [Custodian](https://www.safepal.com/blog/how-scammers-exploit-blockchain-transparency), [MFA](https://safepalsupport.zendesk.com/hc/en-us/articles/360057783771-Passphrase)"
+tokenpocket,!provider:tokenpocket,TRUE,"[Browser Extension"", ""Hardware Wallet"",""iOS"",""Android""]",FALSE,FALSE,TRUE,,,,,,,,,,,,,[Supported Platforms](https://www.tokenpocket.pro/);  [Custodial](https://help.tokenpocket.pro/developer-en); [MSIG](https://www.tokenpocket.pro/)
+coolwallet,!provider:coolwallet,FALSE,"[""Hardware Wallet"",""iOS"",""Android""]",FALSE,TRUE,FALSE,,,,,,,,,,,,,[Open Source](https://www.coolwallet.io/);  [Custodial](https://www.coolwallet.io/blogs/blog/custodial-non-custodial-crypto-wallet?srsltid=AfmBOopctB_QVwqeXqxZ8pELTy0iyNg7kl8ZeviDdSTsWSxkceO1KD8j); [MFA](https://www.coolwallet.io/)
+dcent,!provider:dcent,FALSE,"[""Hardware Wallet"",""iOS"",""Android""]",FALSE,TRUE,FALSE,,,,,,,,,,,,,[Supported Platform](https://userguide.dcentwallet.com/native-service/swap/supported-networks);  [Custodial](https://www.dcentwallet.com/en/expired-recent-policy-stipulations-app); [MFA](https://store.dcentwallet.com/pages/dcent-biometric-crypto-wallet)
+rabby,!provider:rabby,TRUE,"[""Browser Extension"",""iOS"",""Android""]",FALSE,FALSE,FALSE,,,,,,,,,,,,,"[Open Source](https://github.com/RabbyHub/Rabby); [Custodial](https://help.onekey.so/en/articles/11461105-how-to-use-rabby-wallet-with-onekey-hardware-wallets#:~:text=Rabby%20Wallet%20is%20a%20non,have%20garnered%20significant%20user%20appreciation.)"
+trustwallet,!provider:trustwallet,TRUE,"[""Browser Extension"",""iOS"",""Android""]",FALSE,TRUE,FALSE,,,,,,,,,,,,,[Open Source](https://github.com/trustwallet/wallet-core); [Custodial](https://trustwallet.com/blog/cryptocurrency/what-is-a-crypto-wallet) [Supported Platform](https://trustwallet.com/id/blog/guides/trust-wallet-vs-trust-wallet-swift); [MFA](https://trustwallet.com/ethereum-wallet)
+zerion,!provider:zerion,TRUE,"[""Browser Extension"",""iOS"",""Android""]",FALSE,FALSE,FALSE,,,,,,,,,,,,,"[Open Source](https://github.com/zeriontech); [Supported Platform](https://zerion.io/); [Custodial](https://zerion.io/#:~:text=Zerion%20Wallet%20is%20a%20non,opportunities%20across%20DeFi%20and%20NFTs.)"
+phantom,!provider:phantom,TRUE,"[""Browser Extension"",""iOS"",""Android""]",FALSE,TRUE,FALSE,,,,,,,,,,,,,[Open Source](https://github.com/phantom/docs);  [Supported Platform](https://phantom.com/learn/crypto-101/what-is-a-crypto-wallet); [Custodial](https://phantom.com/learn/crypto-101/what-is-a-crypto-wallet); [MFA](https://help.phantom.com/hc/en-us/articles/28951298899603-How-to-enable-device-authentication-in-Phantom-on-mobile)
+unstoppable,!provider:unstoppable,TRUE,"[""iOS"",""Android""]",FALSE,FALSE,FALSE,,,,,,,,,,,,,"[Open Source, Custodial](https://unstoppable.money/)"
```

Comment:
This wallet does not exist in `/providers` folder, so can't be referenced as `!provider`. If this provider is  on only one chain - use his name instead here, if on many chains - move it to `/providers` and refer here.
Example: https://github.com/Chain-Love/chain-love/blob/main/networks/filecoin/wallet.csv
Style guide: https://github.com/Chain-Love/chain-love/wiki/Provider-References

Reply:
Here and on all lines till the end of the file except but lines 17 and 18